sdk/dotnet/Pulumi/Core/Output.cs

```diff
@@ -327,5 +333,12 @@ void Update<X>(OutputData<X> data, ref X location)
                 (isKnown, isSecret) = OutputData.Combine(data, isKnown, isSecret);
             }
         }
+
+        internal static Output<T> Unknown() => Unknown(default!);
```

Comment:
It's a little hard to see syntactically but it seems that the `default!` is not leaking right, the only place we use this is in `Unknown().Apply(_ => valueFactory());`? If we could inline there would be easier on the reader to see that no code ever tries to inspect the `default!`.

Reply:
Done